mounts.go

```diff
@@ -335,6 +335,11 @@ func (taskMount *TaskMount) Start() *CommandExecutionError {
 		// If the problem is internal (e.g. can't mount a writable cache) then
 		// this is handled by a panic.
 		if err != nil {
+			if monitor != nil {
+				incidentID := monitor.ReportError(err, "Error mounting directory")
+				return Failure(fmt.Errorf("%v, incident ID: %s", err, incidentID))
+			}
+
```

Comment:
I think we should only report errors for exceptions, not failures (this would be a task failure, and can be e.g. if an invalid url is specified for the preloaded content of the mount). There should be other places where we explicitly panic in the code.